modules/bootstrapped/src/generated/smithy4s/example/AdtUnionWithTransitiveAndDirectMixins.scala

```diff
@@ -0,0 +1,91 @@
+package smithy4s.example
+
+import smithy4s.Hints
+import smithy4s.Schema
+import smithy4s.ShapeId
+import smithy4s.ShapeTag
+import smithy4s.schema.Schema.long
+import smithy4s.schema.Schema.struct
+import smithy4s.schema.Schema.union
+
+sealed trait AdtUnionWithTransitiveAndDirectMixins extends AdtMixinOne with TransitiveMixin with scala.Product with scala.Serializable { self =>
```

Comment:
This is an interesting one: in both members, `AdtMixinOne` can be reached both directly AND transitively via `TransitiveMixin`. The current implementation adds them both here, which _should_ be OK, although it would be enough for the union to just specify `TransitiveMixin`.

I think this is OK, we're doing something slightly redundant for the benefit of having a simpler implementation in `SmithyToIR`.